geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java

```diff
@@ -360,6 +360,13 @@ private void processRequest(final Socket sock) {
           versionOrdinal = (short) GOSSIP_TO_GEMFIRE_VERSION_MAP.get(gossipVersion);
         } else {
           // Close the socket. We can not accept requests from a newer version
+          try {
+            sock.getOutputStream().write("unknown protocol version".getBytes());
```

Comment:
Is there any risk of this being interpreted by anything other than garbage on the other side?